infra/azure/main.tf

```diff
@@ -475,3 +552,24 @@ module "grafana_sp" {
   application_id = azuread_application.grafana.application_id
   object_id      = azuread_application.grafana.object_id
 }
+
+module "ci" {
+  source = "./ci"
+  count = var.ci_config != null ? 1 : 0
+
+  resource_group        = data.azurerm_resource_group.rg
+  ci_principal_id       = module.ci_sp.principal_id
+  container_registry_id = azurerm_container_registry.acr.id
+}
+
+module "ci_k8s_resources" {
+  source = "../k8s/ci"
```

Comment:
Hmm. So, in terraform, declaring something else as a "module" is similar to invoking/calling that "code"? I guess I'd have expected to declare that k8s "depends on" a certain set of cloud resources. Then, I guess, you'd have to specify which cloud in a variable? Can you comment a bit on why my intuitions are wrong?